src/rgw/rgw_rados.cc

```diff
@@ -1136,6 +1136,8 @@ int RGWRados::register_to_service_map(const string& daemon_type, const map<strin
   if (name.compare(0, 4, "rgw.") == 0) {
     name = name.substr(4);
   }
+  std::string instance_id = stringify(rados.get_instance_id());
+  std::string rgw_service_map_name = name + "." +  metadata["zonegroup_name"] + "." + metadata["zone_name"] + "." + instance_id;
   int ret = rados.service_daemon_register(daemon_type, name, metadata);
```

Comment:
@leseb The rgw_service_map_name need to passed to this function right?

Reply:
> @leseb The `rgw_service_map_name` need to pass to this function right?

Duh! I mixed things up locally 🤦 

> Just out of curiosity, the rbd code passes `name` and `instance_id` to string map of `service_daemon_register() ` but here it is different. Any idea?

I don't know :(